### PR TITLE
Fix entering & exiting vehicle animations

### PR DESCRIPTION
**Pull request by danhedron**
- Don't crash if the seat doesn't have a door
- Make RHS animations count as entering or exiting animations

### CODE REVIEW

File: rwengine/src/objects/CharacterObject.cpp [hunk 1]
```diff
@@ -437,8 +437,11 @@ bool CharacterObject::enterVehicle(VehicleObject* vehicle, size_t seat)
 
 bool CharacterObject::isEnteringOrExitingVehicle() const
 {
-	return animator->getAnimation(AnimIndexAction) == animations.car_getout_lhs ||
-			animator->getAnimation(AnimIndexAction) == animations.car_getin_lhs;
+	auto a = animator->getAnimation(AnimIndexAction);
+	return  a == animations.car_getout_lhs ||
+			a == animations.car_getin_lhs ||
+			a == animations.car_getout_rhs ||
+			a == animations.car_getin_rhs;
```
Comment (JayFoxRox) [inline rwengine/src/objects/CharacterObject.cpp]: nit: alignment gone wrong.

**Comment by JayFoxRox** [comment]
Personally I'd have preferred splitting this PR into 2 commits.
I've skimmed over the code which looks good.

I'll still do a test locally and do a full review soon.

File: rwengine/src/ai/CharacterController.cpp [hunk 2]
```diff
@@ -318,6 +320,8 @@ bool Activities::ExitVehicle::update(CharacterObject *character, CharacterContro
 {
 	RW_UNUSED(controller);
 
+	auto vehicle = character->getCurrentVehicle();
+
```
Comment (JayFoxRox) [inline rwengine/src/ai/CharacterController.cpp]: place `exitPos`, `exitPosLocal` and `isRhs` here?

File: rwengine/src/ai/CharacterController.cpp [hunk 3]
```diff
@@ -353,15 +357,17 @@ bool Activities::ExitVehicle::update(CharacterObject *character, CharacterContro
...
 
-	if( character->getCurrentVehicle() == nullptr ) return true;
+	if( vehicle == nullptr ) return true;
 
-	auto vehicle = character->getCurrentVehicle();
 	auto seat = character->getCurrentSeat();
 	auto door = vehicle->getSeatEntryDoor(seat);
+	auto exitPos = vehicle->getSeatEntryPositionWorld(seat);
+	auto exitPosLocal = vehicle->getSeatEntryPosition(seat);
+	auto isRhs = exitPosLocal.x > 0.f;
```
Comment (JayFoxRox) [inline rwengine/src/ai/CharacterController.cpp]: same as in the jacked case already, move to suggested point above

File: rwengine/src/ai/CharacterController.cpp [hunk 4]
```diff
@@ -329,20 +333,20 @@ bool Activities::ExitVehicle::update(CharacterObject *character, CharacterContro
...
 		else {
-			if (character->getCurrentVehicle() == nullptr) return true;
+			if (vehicle == nullptr) return true;
 
-			auto vehicle = character->getCurrentVehicle();
 			auto seat = character->getCurrentSeat();
-			auto door = vehicle->getSeatEntryDoor(seat);
+			auto exitPos = vehicle->getSeatEntryPositionWorld(seat);
+			auto exitPosLocal = vehicle->getSeatEntryPosition(seat);
+			auto isRhs = exitPosLocal.x > 0.f;
```
Comment (JayFoxRox) [inline rwengine/src/ai/CharacterController.cpp]: get rid of this?

File: rwengine/src/ai/CharacterController.cpp (hunk 2 above)
Comment (danhedron) [inline rwengine/src/ai/CharacterController.cpp]: Vehicle can be null here.

**Comment by JayFoxRox** [comment]
LGTM

Note: Entering boats is super-hard because it warps you inside, however, you still have F pressed so it will instantly cause you to leave again etc. It's beyond the scope of this PR but I felt it's related so I mention it here. Probably deserves it's own issue for boats oddities and boats physics.

**Comment by danhedron** [comment]
I've noticed a few other issues with entering and exiting vehicle caused by the input changes, so that can be noted in an issue about double buffering input to detect presses.

**Pull request merged**